fetch.bs

```diff
@@ -6892,102 +6879,122 @@ returns failure or a <a for=/>MIME type</a>.
+   then:
+
+   <ol>
+    <li>
+     <p>Parse <var>bytes</var>, using the value of the `<code>boundary</code>` parameter from
+     <var>mimeType</var>, per the rules set forth in
+     <cite>Returning Values from Forms: multipart/form-data</cite>. [[!RFC7578]]</p>
+
+     <p>Each part whose `<code>Content-Disposition</code>` header contains a `<code>filename</code>`
+     parameter must be parsed into an <a for="entry list">entry</a> whose value is a {{File}} object
+     whose contents are the contents of the part. The {{File/name}} attribute of the {{File}} object
+     must have the value of the `<code>filename</code>` parameter of the part. The {{Blob/type}}
+     attribute of the {{File}} object must have the value of the `<code>Content-Type</code>` header
+     of the part if the part has such header, and `<code>text/plain</code>` (the default defined by
+     [[!RFC7578]] section 4.4) otherwise.</p>
+
+     <p>Each part whose `<code>Content-Disposition</code>` header does not contain a
+     `<code>filename</code>` parameter must be parsed into an <a for="entry list">entry</a> whose
+     value is the <a lt="UTF-8 decode without BOM">UTF-8 decoded without BOM</a> content of the part.
+     <span class=note>This is done regardless of the presence or the value of a
+     `<code>Content-Type</code>` header and regardless of the presence or the value of a
+     `<code>charset</code>` parameter.</span></p>
+
+     <p class=note>A part whose `<code>Content-Disposition</code>` header contains a
+     `<code>name</code>` parameter whose value is `<code>_charset_</code>` is parsed like any other
+     part. It does not change the encoding.</p>
+    </li>
+
+    <li><p>If that fails for some reason, then run <var>rejectStep</var> with a {{TypeError}} and
+    return.
+
+    <li><p>Run <var>resolveStep</var> with a new {{FormData}} object, appending each
+    <a for="entry list">entry</a>, resulting from the parsing operation, to its
+    <a for=FormData>entry list</a>.
+   </ol>
+
+   <p class=XXX>The above is a rough approximation of what is needed for
+   `<code>multipart/form-data</code>`, a more detailed parsing specification is to be
+   written. Volunteers welcome.
+
+   <p>Otherwise, if <var>mimeType</var>'s <a for="MIME type">essence</a> is
+   "<code>application/x-www-form-urlencoded</code>", then:
+
+   <ol>
+    <li><p>Let <var>entries</var> be the result of
+    <a lt="urlencoded parser">parsing</a> <var>bytes</var>.
+
+    <li><p>If <var>entries</var> is failure, then <a>throw</a> a {{TypeError}}.
+
+    <li><p>Run <var>resolveStep</var> with a new {{FormData}} object whose
+    <a for=FormData>entry list</a> is <var>entries</var>.
+   </ol>
+
+   <p>Otherwise, run <var>rejectStep</var> with a {{TypeError}}.
+
+  <dt><i>JSON</i>
+  <dd><p>Run <var>resolveStep</var> with the result of running <a>parse JSON from bytes</a> on
+  <var>bytes</var>.
```

Comment:
This calls into ["parse a JSON string to a JavaScript value"](https://infra.spec.whatwg.org/#parse-a-json-string-to-a-javascript-value), which can throw if the string is not valid JSON. In that case, we should call `rejectStep`.